### PR TITLE
Stock Notifications

### PR DESCRIPTION
**Pull request by therishidesai**

### CODE REVIEW

File: lib/jobs.rb
Comment (sameetandpotatoes) [inline lib/jobs.rb]: Why don't we just return the items and the locations where there are low quantities of that item as a string, rather than a item json as a string?

That will make it easier when the `groot-notification-service` is up to send this notification, as the message will contain only the useful information.

File: lib/jobs.rb
Comment (sameetandpotatoes) [inline lib/jobs.rb]: NP: replace `items` with `Items.all`

File: app.rb
Comment (sameetandpotatoes) [inline app.rb]: Is there a reason we have to do it on startup, rather than just let the scheduler do it?

File: app.rb
Comment (sameetandpotatoes) [inline app.rb]: Every 3 seconds seems a bit too frequent, why don't we do every 8 hours or something?

@rauhul was also suggesting that we just run this on every vend, not sure which approach we want to go with.

File: app.rb
Comment (sameetandpotatoes) [inline app.rb]: If we're defining the logger here, then I think [this line](https://github.com/acm-uiuc/groot-merch-service/pull/19/files#diff-cc95738088603531796e0d0f246a5d77R53) can be removed, as it would overwrite it?

Can you test and make sure that's the case?

**Review by sameetandpotatoes — DISMISSED** [review]

File: lib/jobs.rb
Comment (apache8080) [reply lib/jobs.rb]: I was just using the string for debugging right now because when the notification service is ready I was thinking of just sending the request from this function.

File: app.rb
Comment (apache8080) [reply app.rb]: Running it on every vend might be better since I don't know how great rufus-scheduler is. As far as 3 seconds that was also for debugging. I was thinking of updating the time to every night but 8 hours would be fine if we decide to use a rufus-scheduler.

File: lib/jobs.rb
Comment (sameetandpotatoes) [reply lib/jobs.rb]: Yeah that's fine, just curious why we would want `i.serialize.to_json` rather than just `i.name`, or `"#{i.name (min(i.vending.map(:quantity))}"` or something like that (something more readable I mean)

File: app.rb
Comment (sameetandpotatoes) [reply app.rb]: I think the better solution is to use a scheduler so that we can make `vending` as fast as possible 

`rufus-scheduler` has a good reputation that works for in-memory scheduling, so I'm not worried about that.

File: lib/jobs.rb
Comment (narendasan) [inline lib/jobs.rb]: Put the URL in a config file and go through the API gateway

**Review by narendasan — DISMISSED** [review]

File: lib/jobs.rb
Comment (narendasan) [reply lib/jobs.rb]: Actually the API gateway URL is already there, just form the endpoint here

File: lib/jobs.rb
Comment (sameetandpotatoes) [inline lib/jobs.rb]: Check that it's a 200, and log an error if it's not.

File: Gemfile
Comment (bcongdon) [inline Gemfile]: Can you add these changed gems to the Gemfile.lock as well? (When you installed them, `bundle` should auto update the lockfile IIRC)

**Comment by apache8080** [comment]
Can you guys approve this PR? Currently everything is fine other than the Gemfile.lock issue which we discussed at last weeks projects meeting as something that shouldn't hold up this PR.

**Review by sameetandpotatoes — DISMISSED** [review]
lgtm other than Gemfile.lock (will this be done in a followup?).

Please rebase with master before merging.

**Review by sameetandpotatoes — APPROVED** [review]
Build passed, merging...

**Pull request merged**
